src/Printer.java

```diff
@@ -0,0 +1,18 @@
+import java.io.PrintStream;
+
+public class Printer {
+    
+    private PrintStream output;
+
+    public Printer() {
+        this.output = System.out;
+    }
+
+    public void printInstruction(String instruction) {
+        output.println(instruction);
+    }
+
+    public void printFeedback(Feedback feedback) {
+        output.println("RESULT: " + feedback.bulls + " BULL and " + feedback.cows + " COW");
```

Comment:
Oh look, a special method for a special event that knows about the format! 
You did this for printing feedback but not printing other game prompts 🤔 

I think if you pick either one and use it consistently, another idea may become clearer.

Lets do this together 💪

Reply:
So my thinking with the Printer is that it should know how to deconstruct the Feedback object and print it on the screen. (I should've done it for the Secret object too but didn't because it wasn't actually a requirement of the system and I was testing my program... 😅)

With the other game prompts, because there was no object involved to pass to the Printer, I thought it would be better to just have a kinda generic method that just prints whatever string it gets. But after reading your comment above, this approach meant that logic of knowing how and what to print is split between Main and Printer. 

That being said, how do we test the other prompts that don't pass in any arguments? Do they need to be tested?